test/helpers/deploy/deploy_resources.go

```diff
@@ -1010,6 +1010,57 @@ func KongDataPlaneClientCertificateAttachedToCP(
 	return cert
 }
 
+// KonnectExtension deploys a KonnectExtension.
+func KonnectExtension(
+	t *testing.T,
+	ctx context.Context,
+	cl client.Client,
+	opts ...ObjOption,
+) *konnectv1alpha1.KonnectExtension {
+	t.Helper()
+
+	ke := &konnectv1alpha1.KonnectExtension{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "konnect-extension-",
+		},
+	}
+	for _, opt := range opts {
+		opt(ke)
+	}
+
+	require.NoError(t, cl.Create(ctx, ke))
+	logObjectCreate(t, ke)
+	return ke
+}
+
+// KonnectExtensionWithAPIAuthRefAndCPID deploys a KonnectExtension attaching to control plane
+// by the CP's ID and the API auth configuration to the Konnect server where the CP is in.
+func KonnectExtensionWithAPIAuthRefAndCPID(
+	t *testing.T,
+	ctx context.Context,
+	cl client.Client,
+	apiAuth konnectv1alpha1.KonnectAPIAuthConfigurationRef,
+	cpID string,
+	opts ...ObjOption,
+) *konnectv1alpha1.KonnectExtension {
+	opts = append(opts, func(obj client.Object) {
+		ke, ok := obj.(*konnectv1alpha1.KonnectExtension)
+		require.Truef(t, ok, "Expect object %s/%s to be a KonnectExtension, actual type %T",
+			obj.GetNamespace(), obj.GetName(), obj)
+		ke.Spec.KonnectConfiguration = &konnectv1alpha1.KonnectConfiguration{
+			APIAuthConfigurationRef: apiAuth,
+		}
+		ke.Spec.KonnectControlPlane.ControlPlaneRef = commonv1alpha1.ControlPlaneRef{
+			Type:      commonv1alpha1.ControlPlaneRefKonnectID,
+			KonnectID: lo.ToPtr(cpID),
+		}
+	})
+	return KonnectExtension(
+		t, ctx, cl,
+		opts...,
+	)
+}
```

Comment:
Wouldn't it be more composable (to be used by other types that support `KonnectConfiguration` field) to define an object option like `WithKonnectConfiguration` and use it similarly to 

```
		created := deploy.KonnectExtension(t, ctx, clientNamespaced,
			deploy.WithKonnectIDControlPlaneRef(cp),
			deploy.WithKonnectConfiguration(konnectConfiguration),
		)
```

WDYT?

Reply:
Extracted `WithKonnectConfiguration`.

Reply:
@pmalek `KonnectExtension` does not support `GetControlPlaneRef`/`SetControlPlaneRef`. It requires updating of `kubernetes-configuartion`. I think we should do it later after https://github.com/Kong/kubernetes-configuration/issues/328.